### ISSUE TITLE
Type Error: [...] ULTIMATE.init must not modify #length

### ISSUE THREAD

**Issue by invd** [ultimate-pa/ultimate#298]
Problem:
`Type Error: Procedure #Ultimate.alloc may modify #length procedure ULTIMATE.init must not modify #length. CallStatement[false,[VariableLHS[#t~string0,null]],#Ultimate.alloc,[BitvecLiteral[261,32]]] calls #Ultimate.alloc. Modifies not transitive`
during translation.

- Minified input: 
[ctrans-bug-min-TE-static-const-uint8_t.c](https://github.com/ultimate-pa/ultimate/blob/dev/trunk/examples/CToBoogieTranslation/regression/ctrans-bug-min-TE-static-const-uint8_t.c)
- Settings: `settings/default/automizer/svcomp-DerefFreeMemtrack-32bit-Automizer_Bitvector.epf`
- Toolchain: `UAutomizer-linux/config/AutomizerMemDerefMemtrack.xml`

This issue occurred during analysis of Trezor firmware and is relevant to progress there.

**Comment by alexandernutz** [comment]
just some documentation:
 - I am gettion an AssertionError in Array ("not outermost"...)
 - a one-dimensional array with a pointer inside is accessed through a two-dimensional access

todos:
 - trigger to move the array on heap
 - make the array access translation tolerant for that case.

**Comment by alexandernutz** [comment]
seems  to work now

**Comment by invd** [comment]
I can confirm that this issue is solved with Ultimate 0.1.23-c168100 for the following settings:
* AutomizerMemDerefMemtrack.xml 
* svcomp-DerefFreeMemtrack-32bit-Automizer_Default.epf as well as  svcomp-DerefFreeMemtrack-32bit-Automizer_Bitvector.epf 

As a side note, Ultimate is still not able to fully verify the minimal example program due to overapproximations, but this was to be expected:
```
  - UnprovableResult [Line: 10]: Unable to prove that pointer dereference always succeeds
    Unable to prove that pointer dereference always succeeds
 Reason: overapproximation of large string literal at line 6.
```

**Comment by invd** [comment]
The underlying issue for the UnprovableResult has been found and fixed.
Ultimate 0.1.23-2874c22 is now able to verify the program:
```

```
